config/registry_image_pruner/image_pruner/prune_images.py

```diff
@@ -87,11 +108,14 @@ def remove_tags(tags: List[Dict[str, Any]], quay_token: str, namespace: str, nam
         # attestation or sbom image
         if (match := tag_regex.match(tag["name"])) is not None:
             if f"sha256:{match.group(1)}" not in image_digests:
-                if dry_run:
-                    LOGGER.info("Tag %s from %s/%s should be removed", tag["name"], namespace, name)
-                else:
-                    LOGGER.info("Removing tag %s from %s/%s", tag["name"], namespace, name)
-                    delete_image_tag(quay_token, namespace, name, tag["name"])
+                # verify that manifest really doesn't exist, because if tag was removed, it won't be tag list, but still be in the registry
+                if not manifest_exists(quay_token, namespace, name, f"sha256:{match.group(1)}"):
+                    if dry_run:
+                        LOGGER.info("Tag %s from %s/%s should be removed", tag["name"], namespace, name)
+                    else:
+                        LOGGER.info("Removing tag %s from %s/%s", tag["name"], namespace, name)
+                        delete_image_tag(quay_token, namespace, name, tag["name"])
```

Comment:
The existent or nonexistent manifest digests can be cached to reduce the number of HTTP requests to Quay.io.

Reply:
+1, currently we're querying the same manifest four times - once for each `sha256-${digest}.(att|sig|sbom|src)`